lib/Frontend/ModuleInterfaceLoader.cpp

```diff
@@ -1910,6 +1931,16 @@ bool ExplicitSwiftModuleLoader::findModule(ImportPath::Element ModuleID,
   }
   auto &moduleInfo = it->getValue();
 
+  // If this is only a Clang module with no paired Swift module, return false
+  // now so that we don't emit diagnostics about it being missing. This gives
+  // ClangImporter an opportunity to import it.
+  bool hasClangModule = !moduleInfo.clangModuleMapPath.empty() ||
```

Comment:
Should this be an `&&`? If the PCM path is not specified, explicit load of this module will fail.

Reply:
AFAICT, `-explicit-swift-module-map-file` doesn't always have to imply explicit Clang modules. The test case `explicit-module-map-clang-implicit.swift` shows that passing only the module map path works, so the implicit Clang module cache is being used there.